prExplorer/client-app/src/app/stores/categoryStore.ts

```diff
@@ -0,0 +1,129 @@
+  }
+
+  get categoriesSortedByName() {
+    const categoryList = Array.from(this.categoryRegistry.values());
+    categoryList.sort((a, b) => {
+      const titleA = a.title.toLowerCase();
+      const titleB = b.title.toLowerCase();
+      return titleA.localeCompare(titleB);
+    });
+    return categoryList;
+  }
+
+  loadCategories = async () => {
+    this.setLoadingInitial(true);
+    try {
+      const categories = await agent.Categories.list();
+      categories.forEach((category) => {
+        this.setCategory(category);
+      });
+      this.setLoadingInitial(false);
+    } catch (error) {
+      console.log(error);
+      this.setLoadingInitial(false);
+    }
+  };
+
+  loadCategory = async (id: string) => {
+    let category = this.getCategory(id);
+    if (category) {
+      this.selectedCategory = category;
+      return category;
+    } else {
+      this.loadingInitial = true;
+      try {
+        category = await agent.Categories.details(id);
+        this.setCategory(category);
+        runInAction(() => this.selectedCategory = category);
+        this.setLoadingInitial(false);
+        return category;
+      } catch (error) {
+        console.log(error);
+        this.setLoadingInitial(false);
+      }
+    }
+  };
+
+  private setCategory = (category: Category) => {
+    this.categoryRegistry.set(category.id, category);
+  };
+
+  private getCategory = (id: string) => {
+    return this.categoryRegistry.get(id);
+  };
+
+  setLoadingInitial = (state: boolean) => {
+    this.loadingInitial = state;
+  };
+
+  createCategory = async (category: Category) => {
+    this.loading = true;
+    category.id = uuid();
+    try {
+      await agent.Categories.create(category);
+      runInAction(() => {
+        this.categoryRegistry.set(category.id, category);
+        this.selectedCategory = category;
+        this.editMode = false;
+        this.loading = false;
+        toast.success("Succesfully Created Category");
+      });
+    } catch (error) {
+      console.log(error);
+      runInAction(() => {
+        this.loading = false;
+      });
+    }
+  };
+
+  updateCategory = async (category: Category) => {
+    this.loading = true;
+    try {
+      await agent.Categories.update(category);
+      runInAction(() => {
+        this.categoryRegistry.set(category.id, category);
+        this.selectedCategory = category;
+        this.editMode = false;
+        this.loading = false;
+        toast.success("Succesfully Edited Category");
+      });
+    } catch (error) {
+      console.log(error);
+      runInAction(() => {
+        this.loading = false;
+      });
+    }
+  };
+
+  deleteCategory = async (id: string) => {
+    this.loading = true;
+    try {
+      await agent.Categories.delete(id);
+      runInAction(() => {
+        this.categoryRegistry.delete(id);
+        this.loading = false;
+        toast.success("Succesfully Deleted Category");
+      });
+    } catch (error) {
+      console.log(error);
```

Comment:
why there still isn't proper and common errors handling implemented here?

Reply:
We'll create new controller on API side for error logs.(Front-Back). Use seperate db (mongoDb cloud) there.